iroh/src/rpc_protocol/authors.rs

```diff
@@ -0,0 +1,146 @@
+use iroh_base::rpc::RpcResult;
+use iroh_docs::{Author, AuthorId};
+use quic_rpc::message::{Msg, RpcMsg, ServerStreaming, ServerStreamingMsg};
+use serde::{Deserialize, Serialize};
+
+use super::RpcService;
+
+#[allow(missing_docs)]
+#[derive(strum::Display, Debug, Serialize, Deserialize)]
+#[nested_enum_utils::enum_conversions(super::Request)]
+pub enum Request {
+    List(ListRequest),
+    Create(CreateRequest),
+    GetDefault(GetDefaultRequest),
+    SetDefault(SetDefaultRequest),
+    Import(ImportRequest),
+    Export(ExportRequest),
+    Delete(DeleteRequest),
+}
+
+#[allow(missing_docs)]
+#[derive(strum::Display, Debug, Serialize, Deserialize)]
+#[nested_enum_utils::enum_conversions(super::Response)]
+pub enum Response {
+    List(RpcResult<ListResponse>),
+    Create(RpcResult<CreateResponse>),
+    GetDefault(RpcResult<GetDefaultResponse>),
+    SetDefault(RpcResult<SetDefaultResponse>),
+    Import(RpcResult<ImportResponse>),
+    Export(RpcResult<ExportResponse>),
+    Delete(RpcResult<DeleteResponse>),
+}
+
+/// List document authors for which we have a secret key.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ListRequest {}
```

Comment:
I wonder if it would read nicer/be a better way to organize these like this:

```rust
pub mod request {
  pub struct List {}
  pub struct Create {}
}

pub mod response {
  pub struct List {}
  pub struct Create {}
}
```

or alternatively 

```rust
pub mod list {
  pub struct Request {}
  pub struct Response {}
}

pub mod create {
  pub struct Request {}
  pub struct Response {}
}
```

Reply:
Nah, I have briefly considered it, but I think you would very frequently have collisions. At the extreme of this approach you don't have any meaning attached to type names at all, and can only glance the meaning by having the complete or at least partial path...

Even removing the subsystem prefixes (NodeFoo to Foo) caused some unfortunate collisions, but my hope is that those will go away once we properly separate the rpc handlers. But for within one subsystem you really want the types to be meaningful by just looking at the local name...